crypto/jwks/component.go

```diff
@@ -0,0 +1,248 @@
+}
+
+// Features returns the features available in this crypto provider.
+func (k *jwksCrypto) Features() []contribCrypto.Feature {
+	return []contribCrypto.Feature{} // No Feature supported.
+}
+
+// Init the JWKS object from the metadata property
+func (k *jwksCrypto) initJWKS(ctx context.Context, md string) error {
+	if len(md) == 0 {
+		return errors.New("metadata property 'jwks' is required")
+	}
+
+	// If the value starts with "http://" or "https://", treat it as URL
+	if strings.HasPrefix(md, "http://") || strings.HasPrefix(md, "https://") {
+		return k.initJWKSFromURL(ctx, md)
+	}
+
+	// Check if the value is a valid path to a local file
+	stat, err := os.Stat(md)
+	if err == nil && stat != nil && !stat.IsDir() {
+		return k.initJWKSFromFile(ctx, md)
+	}
+
+	// Treat the value as the actual JWKS
+	// First, check if it's base64-encoded (remove trailing padding chars if present first)
+	mdJSON, err := base64.RawStdEncoding.DecodeString(strings.TrimRight(md, "="))
+	if err != nil {
+		// Assume it's already JSON, not encoded
+		mdJSON = []byte(md)
+	}
+
+	// Try decoding from JSON
+	k.jwks, err = jwk.Parse(mdJSON)
+	if err != nil {
+		return errors.New("failed to parse metadata property 'jwks': not a URL, path to local file, or JSON value (optionally base64-encoded)")
+	}
+
+	return nil
+}
+
+func (k *jwksCrypto) initJWKSFromURL(ctx context.Context, url string) error {
+	// Create the JWKS cache
+	// We are using k.ctx here because we want this to be tied to the component's lifecycle
+	cache := jwk.NewCache(k.ctx,
+		jwk.WithErrSink(httprc.ErrSinkFunc(func(err error) {
+			k.logger.Warnf("Error while refreshing JWKS cache: %v", err)
+		})),
+	)
+	err := cache.Register(url, jwk.WithMinRefreshInterval(minRefreshInterval))
+	if err != nil {
+		return fmt.Errorf("failed to register JWKS cache: %w", err)
+	}
+
+	// Fetch the JWKS right away to start, so we can check it's valid and populate the cache
+	_, err = cache.Refresh(ctx, url)
+	if err != nil {
+		return fmt.Errorf("failed to fetch JWKS: %w", err)
+	}
+
+	k.jwks = jwk.NewCachedSet(cache, url)
+	return nil
+}
+
+func (k *jwksCrypto) initJWKSFromFile(ctx context.Context, file string) error {
+	// Get the path to the folder containing the file
+	path := filepath.Dir(file)
+
+	// Start watching for changes in the filesystem
+	eventCh := make(chan struct{})
+	loaded := make(chan error)
+	go func() {
+		watchErr := fswatcher.Watch(k.ctx, path, eventCh)
+		if watchErr != nil && !errors.Is(watchErr, context.Canceled) {
+			// Log errors only
+			k.logger.Errorf("Error while watching for changes to the local JWKS file: %v", watchErr)
+		}
+	}()
+	go func() {
+		for {
+			select {
+			case <-eventCh:
+				// When there's a change, reload the JWKS file
+				err := k.parseJWKSFile(file)
+				if loaded != nil {
+					// The first time, signal that the initialization was complete and pass the error
+					loaded <- err
+					close(loaded)
+					loaded = nil
+				} else {
+					// Log errors only
```

Comment:
```suggestion
					// Only log errors
```

Reply:
Actually I think "Log errors only" is more correct, althought it's subtle

- "Only log errors" means -> We will log only in case of error
- "Log errors only" means -> With errors, we'll only show them as logs and not take further action